File: modules/swagger-core/src/test/scala/1_3/converter/RequiredFieldModelTest.scala
package converter

import io.swagger.converter.ModelConverters
import models._
import org.junit.runner.RunWith
import org.scalatest.{FlatSpec, Matchers}
import org.scalatest.junit.JUnitRunner

@RunWith(classOf[JUnitRunner])
class RequiredFieldModelTest extends FlatSpec with Matchers {
  it should "apply read only flag when ApiProperty annotation first" in {
    val models = ModelConverters.getInstance().readAll(classOf[ApiFirstRequiredFieldModel])
    val model = models.get("aaa")
    val prop = model.getProperties().get("a")
    prop.getRequired() should be(true)
  }

  it should "apply read only flag when XmlElement annotation first" in {
    val models = ModelConverters.getInstance().readAll(classOf[XmlFirstRequiredFieldModel])
    val model = models.get("aaa")
    val prop = model.getProperties().get("a")
    prop.getRequired() should be(true)
  }
}

File: modules/swagger-core/src/test/java/io/swagger/RequiredFieldModelTest.java
package io.swagger;

import io.swagger.converter.ModelConverters;
import io.swagger.oas.models.ApiFirstRequiredFieldModel;
import io.swagger.oas.models.XmlFirstRequiredFieldModel;
import io.swagger.oas.models.media.Schema;
import org.testng.annotations.Test;

import java.util.Map;

import static org.testng.Assert.assertTrue;

public class RequiredFieldModelTest {
    @Test(description = "it should apply read only flag when ApiProperty annotation first")
    public void testApiModelPropertyFirstPosition() {
        final Map<String, Schema> models = ModelConverters.getInstance().readAll(ApiFirstRequiredFieldModel.class);
        final Schema  model = models.get("aaa");
        final Schema prop = (Schema)model.getProperties().get("a");
        assertTrue(model.getRequired().contains("a"));
    }

    @Test(description = "it should apply read only flag when XmlElement annotation first")
    public void testApiModelPropertySecondPosition() {
        final Map<String, Schema> models = ModelConverters.getInstance().readAll(XmlFirstRequiredFieldModel.class);
        final Schema model = models.get("aaa");
        final Schema prop = (Schema)model.getProperties().get("a");
        assertTrue(model.getRequired().contains("a"));
    }
}
